### DISCUSSION TITLE
Question related to CLV quickstart

### DISCUSSION THREAD

**Discussion by jacktang** [pymc-labs/pymc-marketing#1008] [Q&A]
Hello,

I am learning CLV tutorial (https://www.pymc-marketing.io/en/stable/notebooks/clv/clv_quickstart.html), and I am confused about below code:

```
customer_1516_history = pd.DataFrame(
    dict(
        customer_id=np.arange(10),
        frequency=np.full(10, customer_1516["frequency"], dtype="int"),
        recency=np.full(10, customer_1516["recency"]),
        T=(np.arange(0, 10) + customer_1516["recency"]).astype("int"),
    )
)
```

Why the `customer_id` is not `customer_1516['customer_id']`. When I set it to `customer_1516['customer_id']`, the code `p_alive = bgm.expected_probability_alive(data=customer_1516_history)` throws duplicated customer_id error. So how can I understand the customer_id should be unique in customer history?

**Comment by jacktang** [comment]
I guess the `customer_id` in dict should be the `transaction_id`(tx_id)?

**Comment by ColtAllen** [comment]
Hey @jacktang,

`customer_1516` is just a single row. If creating a DataFrame from a dict with `customer_id=customer1516['customer_id']`, Pandas will auto-replicate it so it's the same length as the other columns, hence the duplicate error you're getting.

Unique transaction IDs could work in this case, but these are not provided in the raw data. Hope this helps.
